langchain/src/vectorstores/pinecone.ts

```diff
@@ -49,22 +53,24 @@ export class PineconeStore extends VectorStore {
     ids?: string[]
   ): Promise<void> {
     const documentIds = ids == null ? documents.map(() => uuidv4()) : ids;
-    const pineconeVectors = vectors.map((values, idx) => ({
-      id: documentIds[idx],
-      metadata: {
-        ...documents[idx].metadata,
-        [this.textKey]: documents[idx].pageContent,
-      },
-      values,
-    }));
 
-    // Pinecone recommends a limit of 100 vectors per upsert request
-    const chunkSize = 50;
```

Comment:
Thanks for this. We already have this functionality (see the diff around this line). Can you clarify what this improves @danmo